### PR TITLE
Improved folder structure

### PR DESCRIPTION
**Pull request by kwyszyns**
Some things need yet to be corrected:

- SideMenu should be rewritten to new syntax suggested by antd. The current syntax is deprecated and will be out of service in the near future
- translations need to be completed (there are some missing translations)
- I would personally suggest refactoring authQueryWithReAuth function in services/authApi/query.ts file. It is hard to read because of too many nested scopes
- README.md file requires some corrections. I think it would also be useful to provide a documentation in README.md file explaining what kind of code should be placed in particular directories
- there is an error occuring after the app is launched:

image

...it requires further examination

### CODE REVIEW

File: template/src/contexts/ThemeContext.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    </ThemeContext.Provider>
+  );
+};
+
+type ThemeContextData = ReturnType<typeof useProviderData>;
+
+export const useThemeContext = () => {
+  const context = useContext(ThemeContext);
+  if (!context)
+    throw new Error('ThemeContext should be used inside ThemeContextProvider');
```
Comment (skusnierz) [inline template/src/contexts/ThemeContext.tsx:38]: I like it 😄

File: template/src/contexts/ThemeContext.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+interface Props {
+  children: React.ReactNode;
+}
+
+const useProviderData = () => {
+  const [theme, setTheme] = useState<DefaultTheme>(lightTheme);
+
+  const toggleTheme = () => {
+    setTheme((current) => (current === lightTheme ? darkTheme : lightTheme));
```
Comment (skusnierz) [inline template/src/contexts/ThemeContext.tsx:15]: I wonder if we will ever have to support more than 2 themes? Hmm, I think not...

**Review by skusnierz — APPROVED** [review]
LGTM, this folder structure seems more readable to me, thanks @kwyszyns ❤️

File: template/src/contexts/ThemeContext.tsx (hunk 2 above)
Comment (rafalkociatkiewicz) [reply template/src/contexts/ThemeContext.tsx:15]: It's just a toggle. When the need arises to handle more themes, a simple switch will not be sufficient.

**Comment by rafalkociatkiewicz** [comment]
I wonder if we should move to absolute or alias imports in our template.

**Review by rafalkociatkiewicz — APPROVED** [review]

**Pull request merged**